src/index.js

```diff
@@ -184,7 +195,19 @@ exports.createFromBytes = (buf) => {
 }
 
 exports.createFromB58String = (str) => {
-  return new PeerIdWithIs(mh.fromB58String(str))
+  return exports.createFromCID(str) // B58String is CIDv0
+}
+
+exports.createFromCID = (cid) => {
+  if (typeof cid === 'string' || Buffer.isBuffer(cid)) {
+    cid = new CID(cid)
+  } else if (CID.isCID(cid)) {
+    CID.validateCID(cid) // throws on error
```

Comment:
If `CID.isCID` has returned true we _should_ have a valid CID instance. I don't think we need to validate it again do we?!

Reply:
iiuc `isCID` comes from [`class-is`](https://www.npmjs.com/package/class-is) and only checks if `symbolName` is `@ipld/js-cid/CID`